packages/yarnpkg-nm/sources/hoist.ts

```diff
@@ -561,8 +555,13 @@ const getNodeHoistInfo = (rootNode: HoisterWorkTree, rootNodePathLocators: Set<L
               reason = `- peer dependency ${prettyPrintLocator(parentDepNode.locator)} from parent ${prettyPrintLocator(parent.locator)} was not hoisted to ${reasonRoot}`;
             }
           }
+        } else if (
+          !parentDepNode
+          && parent.peerNames.has(name) &&
+          [...node.references].every(a => !parent.references.has(a))
```

Comment:
I doubt it's a good solution to find out if the current node is the same package as the parent